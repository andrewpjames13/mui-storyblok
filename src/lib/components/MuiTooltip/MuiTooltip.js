import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@material-ui/core/Tooltip';
import Storyblok from '../../utils/Storyblok';
import { muiStringProp } from '../../utils/customProps';

const RefComp = React.forwardRef((props, ref) => <div {...props} ref={ref}>{props.children}</div>);

const MuiTooltip = ({
  rootClass,
  title,
  arrow,
  placement,
  children,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);
  return (
    <Tooltip
      data-testid="tooltip"
      className={styles.root}
      placement={placement}
      title={title}
      arrow={arrow}
    >
      <RefComp>
        {children}
      </RefComp>
    </Tooltip>
  );
};

export default MuiTooltip;

MuiTooltip.propTypes = {

  children: PropTypes.node.isRequired,

  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** text for tooltip */
  title: PropTypes.string.isRequired,
  /**
   * mui prop: true | false
   * If true, adds an arrow to the tooltip.
   * */
  arrow: PropTypes.bool,
  /**
   * mui prop:
   'bottom-end' , 'bottom-start' , 'bottom' , 'left-end' , 'left-start'
   , 'left' , 'right-end' , 'right-start' , 'right' , 'top-end' , 'top-start' , 'top'
   * Tooltip placement.
   * */
  placement(props, propName, componentName) {
    const validProps = [
      'bottom-end', 'bottom-start', 'bottom', 'left-end',
      'left-start', 'left', 'right-end', 'right-start',
      'right', 'top-end', 'top-start', 'top',
    ];
    return muiStringProp(props, propName, componentName, validProps);
  },

};

MuiTooltip.defaultProps = {
  arrow: false,
  placement: 'bottom',
  rootClass: [],
};
