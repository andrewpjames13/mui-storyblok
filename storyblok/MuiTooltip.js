module.exports = {
  name: 'MuiTooltip',
  display_name: null,
  updated_at: '2020-03-31T18:27:06.005Z',
  schema: {
    rootClass: {
      type: 'options',
    },
    title: {
      type: 'text',
      required: true,
    },
    arrow: {
      type: 'boolean',
      description: 'f true, adds an arrow to the tooltip.',
    },
    placement: {
      type: 'option',
      use_uuid: true,
      options: [
        {
          value: 'bottom-end',
          name: 'bottom-end',
        },
        {
          value: 'bottom-start',
          name: 'bottom-start',
        },
        {
          value: 'bottom',
          name: 'bottom',
        },
        {
          value: 'left-end',
          name: 'left-end',
        },
        {
          value: 'left-start',
          name: 'left-start',
        },
        {
          value: 'left',
          name: 'left',
        },
        {
          value: 'right-end',
          name: 'right-end',
        },
        {
          value: 'right-start',
          name: 'right-start',
        },
        {
          value: 'right',
          name: 'right',
        },
        {
          value: 'top-end',
          name: 'top-end',
        },
        {
          value: 'top-start',
          name: 'top-start',
        },
        {
          value: 'top',
          name: 'top',
        },
      ],
      required: true,
      default_value: 'bottom',
    },
    content: {
      type: 'bloks',
      restrict_components: true,
      component_whitelist: [
        'MuiIconButtonRedirect',
        'MuiIconButtonHref',
        'MuiIconButtonDownload',
        'MuiIconButtonDialog',
        'MuiButtonRedirect',
      ],
      required: true,
      maximum: '1',
    },
  },
  image: null,
  preview_field: null,
  is_root: false,
  preview_tmpl: null,
  is_nestable: true,
  all_presets: [],
  preset_id: null,
  real_name: 'MuiTooltip',
  component_group_uuid: null,
};
