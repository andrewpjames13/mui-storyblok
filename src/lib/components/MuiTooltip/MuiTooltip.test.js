import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiTooltip from './MuiTooltip';

function setup() {
  const props = {
    title: 'title text',
  };
  const comp = shallow(
    <MuiTooltip {...props}>
      <div>thing here</div>
    </MuiTooltip>,
  );
  return { comp, props };
}

describe('<MuiTooltip />', () => {
  it('renders MuiTooltip', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(
      <MuiTooltip {...props}>
        <div>thing here</div>
      </MuiTooltip>,
    );
    expect(tree).toMatchSnapshot();
  });
});
