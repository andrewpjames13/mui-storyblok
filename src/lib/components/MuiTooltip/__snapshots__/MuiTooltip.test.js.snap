// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTooltip /> snapshot 1`] = `
<button
  aria-describedby={null}
  className="MuiButtonBase-root MuiIconButton-root makeStyles-root-3"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="title text"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root makeStyles-root-27"
      data-testid="muiIconButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root makeStyles-root-28 MuiIcon-colorSecondary"
        >
          star
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
